pkg/service/gcp/kubec.go

```diff
@@ -0,0 +1,156 @@
+package gcp
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/pkg/errors"
+	proto "gitlab.com/netbook-devs/spawner-service/proto/netbookai/spawner"
+	"google.golang.org/genproto/googleapis/container/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+func (g *GCPController) getToken(ctx context.Context, req *proto.GetTokenRequest) (*proto.GetTokenResponse, error) {
+
+	cred, err := getCredentials(ctx, req.AccountName)
+	if err != nil {
+		return nil, errors.Wrap(err, "getToken")
+	}
+	cluster, err := g.getClusterInternal(ctx, cred, req.Region, req.ClusterName)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "GetToken")
+	}
+
+	if cluster.Status != container.Cluster_RUNNING {
+		return nil, fmt.Errorf("cluster is not in running state yet")
+	}
+	server := fmt.Sprintf("https://%s", cluster.Endpoint)
```

Comment:
Can we check if `cluster.Endpoint` does already not have a protocol before prefixing? Parse using `url.Parse` and check using `IsAbs`

Reply:
endpoint is a IP, so must be appeneded. Have checked googles code too(the `gcloud` source). It doesnt have so even they append https while generating kube config.

Reply:
Sure but checking before adding will save us when they change format to give a LB instead of static IP.